java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestSearchUtils.java

```diff
@@ -70,6 +86,102 @@
         return resultList;
     }
 
+    /**
+     * Return a list of test packages and types in the given project. The returned list has the following hierarchy:
```

Comment:
so it's actually a pre-order traversal of the tree?

Reply:
Hmm, what I want to express here is that the elements in returned list might have children